agent/src/util/device_plugin_builder.rs

```diff
@@ -140,8 +140,8 @@ impl DevicePluginBuilderInterface for DevicePluginBuilder {
                     UnixListener::bind(task_socket_path).expect("Failed to bind to socket path");
 
                 async_stream::stream! {
-                    while let item = uds.accept().map_ok(|(st, _)| unix_stream::UnixStream(st)).await {
```

Comment:
i was a little confused handling these. The way this is written it looks like the loop is trying to break the while when there's no more connections to accept ( i think). but according to clippy this is an irrefutable statement that never breaks.
```
warning: irrefutable `while let` pattern
   --> discovery-utils/src/discovery/mod.rs:231:27
    |
231 |                     while let item = uds.accept().map_ok(|(st, _)| unix_stream::UnixStream(st)).await {
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(irrefutable_let_patterns)]` on by default
    = note: this pattern will always match, so the loop will never exit
    = help: consider instead using a `loop { ... }` with a `let` inside it
```
but matching this and yielding the unix stream doesnt work because this fn returns a Result. 

So ultimately i am a little concerned if this is the desired functionality. I trusted clippy and thought id make a note of this.

Reply:
i dont think this causes an infinite loop though. Otherwise any test that was using the [test version of this](https://github.com/project-akri/akri/pull/442/files#diff-87f938518167c58192faa87e9f179c26b5da52546181289fdbd13db1cd36f819R274) would have failed.

Reply:
This loop looks fine -- it almost matches the [tonic example](https://github.com/hyperium/tonic/blob/v0.6.2/examples/src/uds/server.rs#L54). The item from the stream could potentially be stored in an `item` variable [like in the example](https://github.com/hyperium/tonic/blob/v0.6.2/examples/src/uds/server.rs#L54) for better readability.

Reply:
Gotcha, i see what you mean by the tonic example too.
refactored all these to be 
```
let item = ...
yield item;
```
in f333e781c98158c8a462f341b8d002e8de76a37c